extensions/cloudfiles-extensions/src/main/java/io/druid/storage/cloudfiles/CloudFilesDataSegmentPusher.java

```diff
@@ -68,8 +68,9 @@ public String getPathForHadoop(final String dataSource)
   public DataSegment push(final File indexFilesDir, final DataSegment inSegment) throws IOException
   {
     final String segmentPath = CloudFilesUtils.buildCloudFilesPath(this.config.getBasePath(), inSegment);
-    final File zipOutFile = File.createTempFile("druid", "index.zip");
     final File descriptorFile = File.createTempFile("descriptor", ".json");
+    final File zipOutFile = File.createTempFile("druid", "index.zip");
+    final long indexSize = CompressionUtils.zip(indexFilesDir, zipOutFile);
```

Comment:
This should be inside the try to make sure any IOException still properly cleans up zipOutFile